### PR TITLE
Cleanup System.Net.Http tests

### PR DESCRIPTION
**Pull request by davidsh**
* Move more tests to the new test server
* Cleanup 'var' coding patterns
* Remove duplicate tests (*Content clas related) that are already covered with the *Content class tests
* Utiilize the newer test server VerifyUpload endpoint which does 'Content-MD5' verification of transmitted payload to the server as well as payload from the server
* Add GZip related tests in addition to Deflate compression schemes

### CODE REVIEW

**Comment by davidsh** [comment]
@stephentoub @CIPop PTAL

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs [hunk 1]
```diff
@@ -168,15 +168,22 @@ public void MaxRequestContentBufferSize_Set_ThrowsPlatformNotSupportedException(
...
-                HttpResponseMessage response = await client.GetAsync(HttpTestServers.RemoteServerGzipUri);
-                string responseContent = await response.Content.ReadAsStringAsync();
-                Assert.True(responseContent.Contains(DecompressedContentPart));
-                Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                using (HttpResponseMessage response = await client.GetAsync(server))
+                {
+                    Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                    string responseContent = await response.Content.ReadAsStringAsync();
+                    _output.WriteLine(responseContent);                    
+                        TestHelper.VerifyResponseBody(
```
Comment (stephentoub) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: NIt: unnecessary indent

**Comment by stephentoub** [comment]
LGTM

**Comment by davidfowl** [comment]
We need more var in corefx

**Comment by stephentoub** [comment]
> We need more var in corefx

Why?

**Comment by davidfowl** [comment]
Makes the code cleaner. 

```
var s = "This is a string";
```

I'm not sure why you'd need to say it's a string :smile: 

We had a big debate about this on the ASP.NET side and just opted to use var literally everywhere. 

https://github.com/aspnet/Home/wiki/Engineering-guidelines#usage-of-the-var-keyword

/cc @Eilon for more data. I wish we wrote down some more notes.

**Comment by stephentoub** [comment]
We've had debates about it as well and landed on the other side, only using ```var``` when the type is named on the RHS.  I personally do not find your example cleaner, and would rather spend the two extra characters to be crystal clear on the type on the LHS.

**Comment by davidfowl** [comment]
+

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs (hunk 1 above)
Comment (davidsh) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: Fixed.

**Pull request merged**